api/metrics-api-jaxrs/src/main/java/org/hawkular/metrics/api/jaxrs/handler/AvailabilityHandler.java

```diff
@@ -137,11 +136,17 @@ public void createMetric(
     })
     public void getMetrics(
             @Suspended AsyncResponse asyncResponse,
-            @ApiParam(value = "List of tags filters", required = false) @QueryParam("tags") Tags tags) {
+            @ApiParam(value = "List of tags filters", required = false) @QueryParam("tags") Tags tags,
+            @ApiParam(value = "Tags query expression", required = false) @QueryParam("tagsQuery") String tagsQuery) {
 
-        Observable<Metric<AvailabilityType>> metricObservable = (tags == null)
-                ? metricsService.findMetrics(getTenant(), AVAILABILITY)
-                : metricsService.findMetricsWithFilters(getTenant(), AVAILABILITY, tags.getTags());
+        Observable<Metric<AvailabilityType>> metricObservable = null;
+        if (tags != null && tagsQuery == null) {
```

Comment:
Is there any way to easily determine from the format that we are using one format or another? If there could be a conflict where some queries could be matched to both formats, then I think we need to have two different named parameters